### DISCUSSION TITLE
Programmatically create an admin user without using Pocketbase as a framework

### DISCUSSION THREAD

**Discussion by davidhaley** [pocketbase/pocketbase#793] [Q&A]
Without using Pocketbase as a framework, what is the simplest way to programmatically create an admin user?

I would like to script the creation of an admin user in a docker-compose setup.

Thank you!

**Comment by ganigeorgiev** [comment]
Currently you have 2 options:

1. Send a regular HTTP request to `POST /api/admins` via curl or similar (see "Create admin" in https://pocketbase.io/docs/api-admins/). This will work only if the admin you want to create is the first one OR you are already authenticated as another admin (aka. has `Authorization: Admin TOKEN` header).

2. Manually insert the admin record in the `pb_data/data.db` db file using the `sqlite3` cli tool:
```sql
# $ sqlite3 pb_data/data.db

INSERT INTO _admins (id, email, passwordHash, tokenKey, created, updated)
VALUES(
    "sbmbsdb40jyxf7h",
    "test@example.com",
    "$2a$13$1EIwr9jv9bJJxfIUd.EtrOGfXCWAm.NuaFt6ZG6OlWmHSUE1Wwdi.", -- 1234567890 password
    "O4rvW9FSUyTA3xUuQmXR3wHF2db9bHs19nBHeSgVTxerOsTAl4",           -- make sure to change the random key
    "2022-10-17 18:00:00.000",
    "2022-10-17 18:00:00.000"
);
```
_(the `passwordHash` is a bcrypt hash with cost 13 and could be generated online)_

_I'll move the issue to Q&A discussion_

**Reply by ganigeorgiev** [reply]
Console commands to manage the admins (create, delete, password reset, etc.) is in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1) and will be added in the near future.

**Reply by davidhaley** [reply]
Thank you!
